### ISSUE TITLE
[format] clang-format-18 flip-flops on some macro code

### ISSUE THREAD

**Issue by flichtenheld** [llvm/llvm-project#115011]
The following code

```
bool encode_pkcs1() {
#define MAKE_DI(x) {NID_##x, x, sizeof(x)}

  FOO foo[];
}
```

causes clang-format-18 to flip-flop on the formatting. With default options the other state looks like this:

```
bool encode_pkcs1() {
#define MAKE_DI(x)                                                             \
  { NID_##x, x, sizeof(x) }

  FOO foo[];
}
```

It always switches from on to the other code when rerunning clang-format.
I haven't actually found any style or option that makes the problem go away.

This is a regression in clang-format-18, I can't reproduce the problem with -17 or earlier versions.

Interestingly if you change `FOO` to `int` the problem also goes away. Not sure what that means, but the line after the macro definition definitely plays a role here.

**Comment by EugeneZelenko** [comment]
Could you please try 19 or `main` branch?

**Comment by flichtenheld** [comment]
Okay, I did start up an Ubuntu 24.10 container and the problem seems not to be reproducible anymore.
The broken version was `Ubuntu clang-format version 18.1.3 (1ubuntu1)`, but it seems to be fixed in `Ubuntu clang-format version 18.1.8 (11)` and `Ubuntu clang-format version 19.1.1 (1ubuntu1)`.

So I guess this issue can be closed.
